### DISCUSSION TITLE
Why docker is the only way to run photoprism ?

### DISCUSSION THREAD

**Discussion by XDavidT** [photoprism/photoprism#931] [Ideas]
I didn't found any native option to run. I know that docker is easy, and nice, but as long as it keep my storage, can't trust docker to manage my data. I would appricaite if option to native ubuntu install will be added.

**Comment by lastzero** [comment]
We would love that too. However, an app like PhotoPrism has lots of dependencies, and debugging local installation issues requires a significant amount of our time. To speed things up, you may either contribute and help us build a package - or become a sponsor.

**Comment by StephenBrown2** [comment]
> as long as it keep my storage

Docker doesn't keep your storage, you do, and mount the volume so that the docker container can access it. If you destroy/remove the docker container and image, all your data remains the same. Even the database can be mounted as a volume.

**Reply by XDavidT** [reply]
If you are running it in the same machine with your storage. But if my storage in Windows machine ? And my docker machine is a VM ? When thinking about that, why not ask for SMB option ?

**Reply by lastzero** [reply]
So you want us to build a Windows version that includes all dependencies natively?

**Reply by lastzero** [reply]
For mounting network shares via Docker, see https://docs.photoprism.org/getting-started/faq/#how-can-i-mount-network-shares-with-docker

**Reply by XDavidT** [reply]
> 
> 
> So you want us to build a Windows version that includes all dependencies natively?

Never say that. It's open-source, so that was only idea, not request. But after follow your second reply, seem the solution is found !

**Comment by alexisLefebvre** [comment]
> I know that docker is easy, and nice, but as long as it keep my storage, can't trust docker to manage my data

You can use the `PHOTOPRISM_READONLY` value with Docker so that PhotoPrism can't modify the files. Or enable it in settings.

You can also configure the volume with the `ro` flag and PhotoPrism won't be able to modify your files, whatever the readonly mode is configured:

```yaml
    volumes:
      # Your photo and video files ([local path]:[container path]):
      - "/MyPictures:/photoprism/originals:ro"
```
